### DISCUSSION TITLE
Exceptions need to be more specific for retriable errors.

### DISCUSSION THREAD

**Discussion by jgz** [denisneuf/python-amazon-ad-api#135] [Ideas]
Right now no matter what the problem is you will either get AdvertisingApiException or AdvertisingApiForbiddenException.  It looks like in the client it is already determining what kind of error is happening.  Would it be possible to please create specific exceptions at least for errors that can be retried?

Specifically:

- 'Too Many Requests' = AdvertisingApiRequestThrottledException
- 'SERVER_IS_BUSY' = AdvertisingApiTemporarilyUnavailableException

This would make retry//back-off error handling much easier and more in line with how its done in sp_api.

Thanks!

**Comment by denisneuf** [comment]
Well, I will have a look, this are different apis and some errors returns a pure raw html and the response is ok so it is a little bit tricky to get some exceptions depending on the endpoint. When I get some time I will review if some codes can get exceptions in a consistent way. If you have time and want to implement you can create a PR and I will happy to review or help you. Kindest regards.

**Reply by denisneuf** [reply]
I've updated the exceptions, may I ask you if you could test if is working for you?
I made some test endpoints and did not find issues
@jgz
